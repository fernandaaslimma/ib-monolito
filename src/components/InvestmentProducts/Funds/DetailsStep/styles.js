import styled from "styled-components";
import { rem, remFontSize } from "../../../../styles/tools";
import { gray200, gray300 } from "../../../../styles/settings";

export const DetailHeader = styled.div`
  margin: ${rem(0)} ${rem(16)} ${rem(32)} ${rem(16)};
  padding-top: ${rem(27)};
`;
export const FundTypes = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: ${rem(3)};
`;
export const InvestmentType = styled.h3`
  font-family: Lato;
  font-style: normal;
  font-weight: 500;
  font-size: ${remFontSize(16)};
  line-height: 125%;

  letter-spacing: ${rem(0.452308)};

  color: ${gray200};
`;
export const SuitabilityFundType = styled.h3``;

export const FundName = styled.h3`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: ${remFontSize(20)};
  line-height: 125%;
  align-items: left;
  width: ${rem(221)};
  margin-bottom: ${rem(8)};

  letter-spacing: ${rem(0.452308)};

  color: ${gray300};
`;

export const FundDescription = styled.h3`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  font-size: ${remFontSize(14)};
  line-height: 160%;

  padding-top: ${rem(8)};

  letter-spacing: ${rem(0.452308)};

  color: ${gray200};
`;

export const QualifiedInvestorInfo = styled.h3`
  font-family: Lato;
  font-style: normal;
  font-weight: 500;
  line-height: ${rem(20)};
  text-align: left;
  font-size: ${remFontSize(16)};
  letter-spacing: ${rem(0.452308)};
  color: ${gray200};
  padding: 0 ${rem(8)} 0 0;
`;

export const QualifiedInvestorWrapper = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: ${rem(8)};
`;

export const Wrapper = styled.div``;

export const ClickbleIcon = styled.div`
  cursor: pointer;
`;

export const InfoContent = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  font-size: ${remFontSize(14)};
  line-height: 150%;
  letter-spacing: ${rem(0.45)};
  color: ${gray200};
  margin-bottom: ${rem(24)};

  b {
    font-weight: 600;
    color: ${gray300};
  }
`;

export const InfoContentBold = styled(InfoContent)`
  font-weight: 600;
  margin: 0;
  color: ${gray300};
`;

export const InfoWrapper = styled.div`
  display: flex;
  flex-direction: column;

  margin: ${rem(24)} ${rem(16)} ${rem(32)} ${rem(16)};
`;

export const ButtonWrapper = styled.div`
  margin: ${rem(16)} ${rem(-8)} ${rem(-8)} ${rem(-8)};
`;

export const ButtonsWrapper = styled(ButtonWrapper)`
  display: flex;
`;

export const AsteriskMessage = styled.div`
  margin: ${rem(12)} ${rem(8)} ${rem(0)} ${rem(8)};
`;

export const AsteriskTitle = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  font-size: ${rem(14)};
  line-height: 125%;
  color: ${gray200};
`;

export const Content = styled.div`
  margin-bottom: ${rem(24)};
`;

export const InfoContentDetais = styled.div`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  color: ${gray200};
  display: flex;
  flex-direction: column;
  margin-bottom: ${rem(16)};
`;

export const SubTitle = styled.span`
  font-weight: 700;
  font-size: ${remFontSize(16)};
  line-height: ${rem(20)};
  letter-spacing: ${rem(0.45)};
  color: ${gray300};
  margin: 0;
`;

export const ContentWrapper = styled.span`
  font-size: ${remFontSize(14)};
  line-height: ${rem(22, 4)};
  letter-spacing: ${rem(0.45)};
`;

export const ContentWrapper2 = styled.span`
  font-size: ${remFontSize(16)};
  line-height: ${rem(20)};
  letter-spacing: ${rem(0.45)};
  margin-top: ${rem(8)};
`;

export const FirstContentAvailar = styled.span`
  font-size: ${remFontSize(14)};
  line-height: ${rem(17, 5)};
  letter-spacing: ${rem(0.45)};
  display: flex;
  justify-content: space-between;
`;

export const ContentAvailar = styled.span`
  font-size: ${remFontSize(14)};
  line-height: ${rem(17, 5)};
  letter-spacing: ${rem(0.45)};
  display: flex;
  justify-content: space-between;
  margin-top: ${rem(14)};
`;

export const InfoContentResidents = styled.div`
  font-family: Lato;
  font-weight: normal;
  font-style: normal;
  font-size: ${remFontSize(14)};
  line-height: ${rem(22, 4)};
  letter-spacing: ${rem(0.45)};
  color: ${gray200};
`;
