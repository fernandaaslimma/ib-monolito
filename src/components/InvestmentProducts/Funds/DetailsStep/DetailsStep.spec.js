import React from "react";
import { shallow } from "enzyme";
import AnimatedBottonSheet from "../../../common/AnimatedBottomSheet";
import DetailsStep from "./DetailsStep";
import DetailsCard from "../../DetailsCard";
import { Button } from "react-bocombbm-components";
jest.mock("../../../../utils/redirect");
const hardRedirectMock = require("../../../../utils/redirect").redirect;

const API = {
  id: 1,
  name: "BAHIA AM MARAÚ FEEDER BOCOM BBM FIC MULTIMERCADO",
  cnpj: "29.733.842/0001-34",
  description: null,
  manager: "Bahia Asset Management",
  administrator: null,
  minimumBalance: 20000,
  minimumTransaction: 10000,
  initialInvestment: 20000,
  riskProfile: "Moderate",
  riskProfileLabel: "Moderado",
  allowedSubscriptions: false,
  qualifiedInvestor: true,
  notAllowedSubscriptionsMessage:
    "Fundo só permite aplicações por investidor qualificado.",
  returns: {
    twelveMonths: null,
    year: null,
    thirtySixMonths: null,
    fortyEightMonths: null
  },
  class: "Hedge Funds",
  classLabel: "Multimercado",
  subscription: {
    type: "Corrido",
    conversionDays: 0,
    settlementDays: 0
  },
  redemption: {
    type: "Corrido",
    conversionDays: 30,
    settlementDays: 1
  },
  benchmark: "CDI",
  quotaDate: "2020-12-21",
  administrationFee: null,
  performanceFee: null,
  inceptionDate: null,
  monthActivity: 12
};
const thisProps = {
  stepForward: jest.fn(),
  currentStep: 2,
  stepBack: jest.fn()
};

const props = {
  handleUserInputTransferCurrency: jest.fn(),
  availableDateRanges: [
    { maxEndTime: 1610060400000, minStartTime: 1610024400000 }
  ],
  authFactors: [
    {
      id: "e80857d6-8f81-408d-84b8-90f49b302da7",
      defaultAuth: true,
      authUri: "pj_franciscogoncalez@bocombbm.com.br",
      type: "mail",
      actions: ["approveinvestment"],
      activated: true,
      plataformIdentifier: null,
      approved: true,
      isSelf: true
    },
    {
      id: "8c82c0e1-12c4-4ba2-b9b0-b3c012651234",
      defaultAuth: false,
      authUri: "mobile",
      type: "mobile",
      actions: ["wiretransfer", "passwordreset", "approvesuitability"],
      activated: true,
      plataformIdentifier: null,
      approved: true,
      isSelf: false
    }
  ]
};

const props2 = {
  handleUserInputTransferCurrency: jest.fn(),
  availableDateRanges: [
    { maxEndTime: 1610060400000, minStartTime: 1610024400000 }
  ],
  authFactors: [
    {
      actions: []
    },
    {
      actions: ["wiretransfer", "passwordreset", "approvesuitability"]
    }
  ]
};

const colors = {
  moderate: "#E3F1D4",
  aggressive: "#E2DCF5",
  conservative: "#DAE6F2"
};

const context = {
  props,
  state: {
    selectedFund: API,
    typeOfFundPendencie: "Ficha Cadastral",
    route: "/personRegistration",
    canInvest: true,
    monthActivity: 11
  },
  colors,
  selectFund: jest.fn(),
  verifyPendencies: jest.fn(),
  resetInvestability: jest.fn(),
  setChosenOperation: jest.fn()
};

const context2 = {
  props,
  state: { selectedFund: { ...API, monthActivity: 11 } },
  colors,
  selectFund: jest.fn(),
  resetInvestability: jest.fn(),
  setChosenOperation: jest.fn()
};

const context3 = {
  props,
  state: { selectedFund: null },
  colors,
  selectFund: jest.fn(),
  resetInvestability: jest.fn(),
  setChosenOperation: jest.fn()
};

const context4 = {
  props: props2,
  state: {
    selectedFund: API,
    typeOfFundPendencie: "Questionário de Suitability",
    route: "suitability",
    canInvest: true
  },
  colors,
  selectFund: jest.fn(),
  verifyPendencies: jest.fn(),
  resetInvestability: jest.fn(),
  setChosenOperation: jest.fn()
};

describe("DetailsStep component", () => {
  let setState, useStateSpy, useEffect;
  const mockUseEffect = () => {
    useEffect.mockImplementation(f => f());
  };
  beforeEach(() => {
    setState = jest.fn();
    useStateSpy = jest.spyOn(React, "useState");
    useStateSpy.mockImplementation(init => [init, setState]);
    React.useContext = jest.fn(() => context);

    useEffect = jest.spyOn(React, "useEffect");
    mockUseEffect();
  });

  it.skip("Should match snapshot", () => {
    const component = shallow(<DetailsStep {...thisProps} />);
    expect(component).toMatchSnapshot();
  });

  it.skip("Should match snapshot with monthActivity less than 12", () => {
    React.useContext = jest.fn(() => context2);
    const component = shallow(<DetailsStep {...thisProps} />);
    expect(component).toMatchSnapshot();
  });

  it.skip("Should match snapshot without selectedFund", () => {
    React.useContext = jest.fn(() => context3);
    const component = shallow(<DetailsStep {...thisProps} />);
    expect(component).toMatchSnapshot();
  });

  it.skip("Should match snapshot without authFactor to invest", () => {
    React.useContext = jest.fn(() => context4);
    const component = shallow(<DetailsStep {...thisProps} />);
    expect(component).toMatchSnapshot();
  });

  it.skip("Should open information", () => {
    const component = shallow(<DetailsStep {...thisProps} />);
    component
      .find(DetailsCard)
      .at(0)
      .prop("clickInfo")();

    expect(setState).toHaveBeenCalled();
  });

  it.skip("Should close information", () => {
    const component = shallow(<DetailsStep {...thisProps} />);
    component
      .find(AnimatedBottonSheet)
      .at(0)
      .prop("onClickInBack")();

    expect(setState).toHaveBeenCalled();
  });

  it.skip("Should open pendencie information", () => {
    const component = shallow(<DetailsStep {...thisProps} />);
    component
      .find(Button)
      .at(1)
      .simulate("click");

    expect(setState).toHaveBeenCalled();
  });

  it.skip("Should close pendencie information", () => {
    const component = shallow(<DetailsStep {...thisProps} />);
    component
      .find(AnimatedBottonSheet)
      .at(1)
      .prop("onClickInBack")();

    expect(setState).toHaveBeenCalled();
  });

  it.skip("Should close pendencie information", () => {
    const component = shallow(<DetailsStep {...thisProps} />);
    component
      .find(Button)
      .at(2)
      .simulate("click");

    expect(setState).toHaveBeenCalled();
  });

  it.skip("Should redirect", () => {
    const component = shallow(<DetailsStep {...thisProps} />);
    component
      .find(Button)
      .at(5)
      .simulate("click");

    expect(hardRedirectMock).toHaveBeenCalledWith(context.state.route);
  });

  it.skip("Should close IorIof information", () => {
    const component = shallow(<DetailsStep {...thisProps} />);
    component
      .find(AnimatedBottonSheet)
      .at(3)
      .prop("onClickInBack")();

    expect(setState).toHaveBeenCalled();
  });

  it.skip("Should close IorIof information", () => {
    const component = shallow(<DetailsStep {...thisProps} />);
    component
      .find(Button)
      .at(6)
      .simulate("click");

    expect(setState).toHaveBeenCalled();
  });
});
