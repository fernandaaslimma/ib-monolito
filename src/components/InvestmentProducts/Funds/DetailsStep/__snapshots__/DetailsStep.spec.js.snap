// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsStep component Should match snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.span
        id="DetailsStep"
      />
      <styled.div>
        <styled.div>
          <styled.h3 />
          <Tag
            color="#E3F1D4"
            title="Moderado"
          />
        </styled.div>
        <styled.h3>
          BAHIA AM MARAÚ FEEDER BOCOM BBM FIC MULTIMERCADO
        </styled.h3>
        <styled.div>
          <styled.h3 />
          <styled.span
            onClick={[Function]}
          >
            <Icon
              color="#003b77"
              height={16}
              spacing={
                Object {
                  "bottom": "none",
                  "left": "none",
                  "right": "none",
                  "top": "none",
                }
              }
              type="Interrogation"
              width={16}
            />
          </styled.span>
        </styled.div>
        <styled.h3 />
      </styled.div>
      <AlertMessage
        dataTest="alertMessage"
        icon="Attention"
        spacing={
          Object {
            "bottom": "l",
            "left": "s",
            "right": "s",
            "top": "none",
          }
        }
        type="neutral"
      >
        <Styled(styled.span)>
          01:01 AM
          10:01 PM
        </Styled(styled.span)>
      </AlertMessage>
      <DetailsCard
        clickInfo={[Function]}
        dataTest="profitabilityDetailsCard"
        grossResultBalance={null}
        list={
          Object {
            "fortyEightMonths": null,
            "thirtySixMonths": null,
            "twelveMonths": null,
            "twentyFourMonths": undefined,
            "yearFunds": null,
          }
        }
        netIncome={null}
        periodActive={12}
        profitabilitySoFar={null}
        title={null}
      />
      <DetailsCard
        dataTest="generalInfoCard"
        list={
          Object {
            "initialInvestment": 20000,
            "minimumBalance": 20000,
            "minimumTransaction": 10000,
            "redempConversionDays": 30,
            "redempSettlementDays": 1,
            "subsConversionDays": 0,
          }
        }
      />
      <DetailsCard
        dataTest="detailsCard"
        list={
          Object {
            "administrationFeeFunds": null,
            "administratorFunds": null,
            "benchmarkFunds": "CDI",
            "custodianFunds": undefined,
            "inceptionDateFunds": null,
            "managerFunds": "Bahia Asset Management",
            "performanceFeeFunds": null,
          }
        }
      />
      <styled.div>
        <styled.span
          data-test="notAllowedSubscriptionsMessage"
        >
          *Fundo só permite aplicações por investidor qualificado.
        </styled.span>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.span />
      <styled.div>
        <Button
          dataTest="Button"
          disabled={false}
          onClick={[MockFunction]}
          spacing={
            Object {
              "bottom": "s",
              "left": "s",
              "right": "xxs",
              "top": "s",
            }
          }
          type="outline"
        />
        <Button
          dataTest="investFund"
          disabled={true}
          onClick={[Function]}
          spacing={
            Object {
              "bottom": "s",
              "left": "xxs",
              "right": "s",
              "top": "s",
            }
          }
        />
      </styled.div>
    </styled.div>
    <AnimatedBottonSheet
      head={
        Object {
          "title": undefined,
        }
      }
      isOpen={false}
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div>
        <styled.div>
          <styled.span
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
          />
          <styled.div>
            <Button
              dataTest="Button"
              disabled={false}
              onClick={[Function]}
              spacing={
                Object {
                  "bottom": "xxs",
                  "left": "xxs",
                  "right": "xxs",
                  "top": "xxs",
                }
              }
              type="outline"
            />
          </styled.div>
        </styled.div>
      </styled.div>
    </AnimatedBottonSheet>
    <AnimatedBottonSheet
      head={
        Object {
          "title": undefined,
        }
      }
      isOpen={false}
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div>
        <styled.div>
          <styled.span>
            <Styled(styled.span)>
              05/31/2010
              .
            </Styled(styled.span)>
          </styled.span>
          <styled.div>
            <Button
              dataTest="Button"
              disabled={false}
              onClick={[Function]}
              spacing={
                Object {
                  "bottom": "xxs",
                  "left": "xxs",
                  "right": "xxs",
                  "top": "xxs",
                }
              }
              type="outline"
            />
          </styled.div>
        </styled.div>
      </styled.div>
    </AnimatedBottonSheet>
    <AnimatedBottonSheet
      head={
        Object {
          "title": undefined,
        }
      }
      isOpen={false}
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div>
        <styled.div>
          <styled.span
            data-test="typeOfFundPendencie"
          >
            <Styled(styled.span) />
          </styled.span>
          <Styled(styled.div)>
            <styled.div>
              <Button
                dataTest="noButton"
                disabled={false}
                onClick={[Function]}
                spacing={
                  Object {
                    "bottom": "xxs",
                    "left": "xxs",
                    "right": "xxs",
                    "top": "xxs",
                  }
                }
                type="outline"
              />
            </styled.div>
            <styled.div>
              <Button
                dataTest="yesButton"
                disabled={false}
                onClick={[Function]}
                spacing={
                  Object {
                    "bottom": "xxs",
                    "left": "xxs",
                    "right": "xxs",
                    "top": "xxs",
                  }
                }
              />
            </styled.div>
          </Styled(styled.div)>
        </styled.div>
      </styled.div>
    </AnimatedBottonSheet>
    <AnimatedBottonSheet
      head={
        Object {
          "title": undefined,
        }
      }
      isOpen={false}
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div>
        <styled.div>
          <styled.div>
            <styled.span />
          </styled.div>
          <styled.div>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
          </styled.div>
          <styled.div>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
          </styled.div>
          <styled.div>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
          </styled.div>
          <styled.div>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
          </styled.div>
          <styled.div>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
          </styled.div>
          <styled.div />
          <Styled(styled.div)>
            <styled.div>
              <Button
                dataTest="noButton"
                disabled={false}
                onClick={[Function]}
                spacing={
                  Object {
                    "bottom": "xxs",
                    "left": "xxs",
                    "right": "xxs",
                    "top": "xxs",
                  }
                }
                type="outline"
              />
            </styled.div>
          </Styled(styled.div)>
        </styled.div>
      </styled.div>
    </AnimatedBottonSheet>
  </styled.div>
</styled.div>
`;

exports[`DetailsStep component Should match snapshot with monthActivity less than 12 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.span
        id="DetailsStep"
      />
      <styled.div>
        <styled.div>
          <styled.h3 />
          <Tag
            color="#E3F1D4"
            title="Moderado"
          />
        </styled.div>
        <styled.h3>
          BAHIA AM MARAÚ FEEDER BOCOM BBM FIC MULTIMERCADO
        </styled.h3>
        <styled.div>
          <styled.h3 />
          <styled.span
            onClick={[Function]}
          >
            <Icon
              color="#003b77"
              height={16}
              spacing={
                Object {
                  "bottom": "none",
                  "left": "none",
                  "right": "none",
                  "top": "none",
                }
              }
              type="Interrogation"
              width={16}
            />
          </styled.span>
        </styled.div>
        <styled.h3 />
      </styled.div>
      <AlertMessage
        dataTest="alertMessage"
        icon="Attention"
        spacing={
          Object {
            "bottom": "l",
            "left": "s",
            "right": "s",
            "top": "none",
          }
        }
        type="neutral"
      >
        <Styled(styled.span)>
          07:01 PM
          04:01 PM
        </Styled(styled.span)>
      </AlertMessage>
      <DetailsCard
        clickInfo={[Function]}
        dataTest="profitabilityDetailsCard"
        grossResultBalance={null}
        list={
          Object {
            "fortyEightMonths": null,
            "thirtySixMonths": null,
            "twelveMonths": null,
            "twentyFourMonths": undefined,
            "yearFunds": null,
          }
        }
        netIncome={null}
        periodActive={11}
        title={null}
      />
      <DetailsCard
        dataTest="generalInfoCard"
        list={
          Object {
            "initialInvestment": 20000,
            "minimumBalance": 20000,
            "minimumTransaction": 10000,
            "redempConversionDays": 30,
            "redempSettlementDays": 1,
            "subsConversionDays": 0,
          }
        }
      />
      <DetailsCard
        dataTest="detailsCard"
        list={
          Object {
            "administrationFeeFunds": null,
            "administratorFunds": null,
            "benchmarkFunds": "CDI",
            "custodianFunds": undefined,
            "inceptionDateFunds": null,
            "managerFunds": "Bahia Asset Management",
            "performanceFeeFunds": null,
          }
        }
      />
      <styled.div>
        <styled.span
          data-test="notAllowedSubscriptionsMessage"
        >
          *Fundo só permite aplicações por investidor qualificado.
        </styled.span>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.span />
      <styled.div>
        <Button
          dataTest="Button"
          disabled={false}
          onClick={[MockFunction]}
          spacing={
            Object {
              "bottom": "s",
              "left": "s",
              "right": "xxs",
              "top": "s",
            }
          }
          type="outline"
        />
        <Button
          dataTest="investFund"
          disabled={true}
          onClick={[Function]}
          spacing={
            Object {
              "bottom": "s",
              "left": "xxs",
              "right": "s",
              "top": "s",
            }
          }
        />
      </styled.div>
    </styled.div>
    <AnimatedBottonSheet
      head={
        Object {
          "title": undefined,
        }
      }
      isOpen={false}
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div>
        <styled.div>
          <styled.span
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
          />
          <styled.div>
            <Button
              dataTest="Button"
              disabled={false}
              onClick={[Function]}
              spacing={
                Object {
                  "bottom": "xxs",
                  "left": "xxs",
                  "right": "xxs",
                  "top": "xxs",
                }
              }
              type="outline"
            />
          </styled.div>
        </styled.div>
      </styled.div>
    </AnimatedBottonSheet>
    <AnimatedBottonSheet
      head={
        Object {
          "title": undefined,
        }
      }
      isOpen={false}
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div>
        <styled.div>
          <styled.span>
            <Styled(styled.span)>
              05/31/2010
              .
            </Styled(styled.span)>
          </styled.span>
          <styled.div>
            <Button
              dataTest="Button"
              disabled={false}
              onClick={[Function]}
              spacing={
                Object {
                  "bottom": "xxs",
                  "left": "xxs",
                  "right": "xxs",
                  "top": "xxs",
                }
              }
              type="outline"
            />
          </styled.div>
        </styled.div>
      </styled.div>
    </AnimatedBottonSheet>
    <AnimatedBottonSheet
      head={
        Object {
          "title": undefined,
        }
      }
      isOpen={false}
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div>
        <styled.div>
          <styled.span
            data-test="typeOfFundPendencie"
          >
            <Styled(styled.span) />
          </styled.span>
          <Styled(styled.div)>
            <styled.div>
              <Button
                dataTest="noButton"
                disabled={false}
                onClick={[Function]}
                spacing={
                  Object {
                    "bottom": "xxs",
                    "left": "xxs",
                    "right": "xxs",
                    "top": "xxs",
                  }
                }
                type="outline"
              />
            </styled.div>
            <styled.div>
              <Button
                dataTest="yesButton"
                disabled={false}
                onClick={[Function]}
                spacing={
                  Object {
                    "bottom": "xxs",
                    "left": "xxs",
                    "right": "xxs",
                    "top": "xxs",
                  }
                }
              />
            </styled.div>
          </Styled(styled.div)>
        </styled.div>
      </styled.div>
    </AnimatedBottonSheet>
    <AnimatedBottonSheet
      head={
        Object {
          "title": undefined,
        }
      }
      isOpen={false}
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div>
        <styled.div>
          <styled.div>
            <styled.span />
          </styled.div>
          <styled.div>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
          </styled.div>
          <styled.div>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
          </styled.div>
          <styled.div>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
          </styled.div>
          <styled.div>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
          </styled.div>
          <styled.div>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
          </styled.div>
          <styled.div />
          <Styled(styled.div)>
            <styled.div>
              <Button
                dataTest="noButton"
                disabled={false}
                onClick={[Function]}
                spacing={
                  Object {
                    "bottom": "xxs",
                    "left": "xxs",
                    "right": "xxs",
                    "top": "xxs",
                  }
                }
                type="outline"
              />
            </styled.div>
          </Styled(styled.div)>
        </styled.div>
      </styled.div>
    </AnimatedBottonSheet>
  </styled.div>
</styled.div>
`;

exports[`DetailsStep component Should match snapshot without authFactor to invest 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.span
        id="DetailsStep"
      />
      <styled.div>
        <styled.div>
          <styled.h3 />
          <Tag
            color="#E3F1D4"
            title="Moderado"
          />
        </styled.div>
        <styled.h3>
          BAHIA AM MARAÚ FEEDER BOCOM BBM FIC MULTIMERCADO
        </styled.h3>
        <styled.div>
          <styled.h3 />
          <styled.span
            onClick={[Function]}
          >
            <Icon
              color="#003b77"
              height={16}
              spacing={
                Object {
                  "bottom": "none",
                  "left": "none",
                  "right": "none",
                  "top": "none",
                }
              }
              type="Interrogation"
              width={16}
            />
          </styled.span>
        </styled.div>
        <styled.h3 />
      </styled.div>
      <AlertMessage
        dataTest="alertMessage"
        icon="Attention"
        spacing={
          Object {
            "bottom": "l",
            "left": "s",
            "right": "s",
            "top": "none",
          }
        }
        type="neutral"
      >
        <Styled(styled.span)>
          01:01 PM
          10:01 AM
        </Styled(styled.span)>
      </AlertMessage>
      <DetailsCard
        clickInfo={[Function]}
        dataTest="profitabilityDetailsCard"
        grossResultBalance={null}
        list={
          Object {
            "fortyEightMonths": null,
            "thirtySixMonths": null,
            "twelveMonths": null,
            "twentyFourMonths": undefined,
            "yearFunds": null,
          }
        }
        netIncome={null}
        periodActive={12}
        profitabilitySoFar={null}
        title={null}
      />
      <DetailsCard
        dataTest="generalInfoCard"
        list={
          Object {
            "initialInvestment": 20000,
            "minimumBalance": 20000,
            "minimumTransaction": 10000,
            "redempConversionDays": 30,
            "redempSettlementDays": 1,
            "subsConversionDays": 0,
          }
        }
      />
      <DetailsCard
        dataTest="detailsCard"
        list={
          Object {
            "administrationFeeFunds": null,
            "administratorFunds": null,
            "benchmarkFunds": "CDI",
            "custodianFunds": undefined,
            "inceptionDateFunds": null,
            "managerFunds": "Bahia Asset Management",
            "performanceFeeFunds": null,
          }
        }
      />
      <styled.div>
        <styled.span
          data-test="notAllowedSubscriptionsMessage"
        >
          *Fundo só permite aplicações por investidor qualificado.
        </styled.span>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.span />
      <styled.div>
        <Button
          dataTest="Button"
          disabled={false}
          onClick={[MockFunction]}
          spacing={
            Object {
              "bottom": "s",
              "left": "s",
              "right": "xxs",
              "top": "s",
            }
          }
          type="outline"
        />
        <Button
          dataTest="investFund"
          disabled={true}
          onClick={[Function]}
          spacing={
            Object {
              "bottom": "s",
              "left": "xxs",
              "right": "s",
              "top": "s",
            }
          }
        />
      </styled.div>
    </styled.div>
    <AnimatedBottonSheet
      head={
        Object {
          "title": undefined,
        }
      }
      isOpen={false}
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div>
        <styled.div>
          <styled.span
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
          />
          <styled.div>
            <Button
              dataTest="Button"
              disabled={false}
              onClick={[Function]}
              spacing={
                Object {
                  "bottom": "xxs",
                  "left": "xxs",
                  "right": "xxs",
                  "top": "xxs",
                }
              }
              type="outline"
            />
          </styled.div>
        </styled.div>
      </styled.div>
    </AnimatedBottonSheet>
    <AnimatedBottonSheet
      head={
        Object {
          "title": undefined,
        }
      }
      isOpen={false}
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div>
        <styled.div>
          <styled.span>
            <Styled(styled.span)>
              05/31/2010
              .
            </Styled(styled.span)>
          </styled.span>
          <styled.div>
            <Button
              dataTest="Button"
              disabled={false}
              onClick={[Function]}
              spacing={
                Object {
                  "bottom": "xxs",
                  "left": "xxs",
                  "right": "xxs",
                  "top": "xxs",
                }
              }
              type="outline"
            />
          </styled.div>
        </styled.div>
      </styled.div>
    </AnimatedBottonSheet>
    <AnimatedBottonSheet
      head={
        Object {
          "title": undefined,
        }
      }
      isOpen={false}
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div>
        <styled.div>
          <styled.span
            data-test="typeOfFundPendencie"
          >
            <Styled(styled.span) />
          </styled.span>
          <Styled(styled.div)>
            <styled.div>
              <Button
                dataTest="noButton"
                disabled={false}
                onClick={[Function]}
                spacing={
                  Object {
                    "bottom": "xxs",
                    "left": "xxs",
                    "right": "xxs",
                    "top": "xxs",
                  }
                }
                type="outline"
              />
            </styled.div>
            <styled.div>
              <Button
                dataTest="yesButton"
                disabled={false}
                onClick={[Function]}
                spacing={
                  Object {
                    "bottom": "xxs",
                    "left": "xxs",
                    "right": "xxs",
                    "top": "xxs",
                  }
                }
              />
            </styled.div>
          </Styled(styled.div)>
        </styled.div>
      </styled.div>
    </AnimatedBottonSheet>
    <AnimatedBottonSheet
      head={
        Object {
          "title": undefined,
        }
      }
      isOpen={false}
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div>
        <styled.div>
          <styled.div>
            <styled.span />
          </styled.div>
          <styled.div>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
          </styled.div>
          <styled.div>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
          </styled.div>
          <styled.div>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
          </styled.div>
          <styled.div>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
          </styled.div>
          <styled.div>
            <styled.span>
              <Styled(styled.span) />
            </styled.span>
          </styled.div>
          <styled.div />
          <Styled(styled.div)>
            <styled.div>
              <Button
                dataTest="noButton"
                disabled={false}
                onClick={[Function]}
                spacing={
                  Object {
                    "bottom": "xxs",
                    "left": "xxs",
                    "right": "xxs",
                    "top": "xxs",
                  }
                }
                type="outline"
              />
            </styled.div>
          </Styled(styled.div)>
        </styled.div>
      </styled.div>
    </AnimatedBottonSheet>
  </styled.div>
</styled.div>
`;

exports[`DetailsStep component Should match snapshot without selectedFund 1`] = `""`;
